### DISCUSSION TITLE
Notification when the car or window is open for an extended period of time

### DISCUSSION THREAD

**Discussion by rtega** [teslamate-org/teslamate#698] [Ideas]
I don't know why the tesla-app doesn't do this in the first place. I guess they don't want to be liable when people get their car or items from it stolen when the app by accident doesn't notify them.

The idea is to have teslamate send an email or some other notification when it notices that one of the following states happens for "an extended period of time":
- the car is unlocked
- one or more doors is open
- one or more windows is open
- the frunk and/or the trunk is open

The extended period of time in minutes would be an option in the settings.

Is this doable?

**Comment by cleanev** [comment]
There is an app on iOS named stats that does this exact thing that you mention. Unfortunately there is no android equivalent

**Comment by baylanger** [comment]
Configuring mail service can be tricky.  Most people would configure their ISP or VM provider's mail server and that often requires a username/password.  The SMPT service might be on port 25 and if it's using encryption that's another port.

I'd too like TM to be able to send me notification or automate some tasks.  I believe there are a few online services that can probably notify you via a smartapp, seems I recall seeing something in the past but can't remember the name.  This requires also to install and configure some software in a docker instance.  It's definitely possible to do.  Hopefully some people will comment below ideas how to do this "easily" for everyone.

**Comment by ngardiner** [comment]
The window state is exported by MQTT, so any system which can consume MQTT topics could likely do this. In homeassistant for example, you could set an automation which triggers a script with a delay every time the window state is down. If the window state is set back to up, it would cancel the script. Once the delay has passed, you can use the notification components such as Amazon SNS (for email), Telegram, even Facebook Messenger to send a message.

If OP is using HASS outside of TeslaMate, I would be happy to put together an example automation for this, as I think it would be handy to have myself. We already have some HASS integration examples as part of the doco so if I get a chance to do this, I'll contribute it.

**Comment by rtega** [comment]
Didn't know it could be that simple with mosquitto_sub. I hashed a simple bash-script to do this which could be further improved upon. I run it each 10 minutes with crontab so it is bound to catch some false positives but it works.
[checkscript.zip](https://github.com/adriankumpf/teslamate/files/4605269/checkscript.zip)

**Comment by rtega** [comment]
You could also use something like Wirepusher to send a notification to your Android phone via curl


```
curl "https://wirepusher.com/send?id=your_id&title=Alert&message=`cat /tmp/doorswindowstate|tr '\n' ' '`&type=default"

```

**Comment by JakobLichterfeld** [comment]
As of SW version 2020.32 this is build into Tesla, notification with Mobile App, so no need for own workarounds anymore :-)

**Comment by rtega** [comment]
Indeed, but for the moment it turns out to be not so reliable.

**Comment by denisjoshua** [comment]
Hi there all, 
sorry if I insert my request in this post... but I'm near to the subject.

First of all thanks so much for this wonderful software that in my opinion is the best. 
A already make a lot of automation using the MQTT topics from TeslaMate. 

Now I just work In Node-Red to a flow that already send me a sample email when 
Tesla arrive Home (using "geofence" topic).
What I want to ask is:
- how do I know that the Tesla finish a drive ? 
- can I extract a pic of the drive in order to put it in the email ? 

What I want is to receive an email like TeslaFI with pic and statistics if possible. 

Thanks a lot all 
Denis
